doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,135 @@
+1) The archive contain only one binary that need do to be extracted and add to PATH env var
+2) The archive contain an installer that need to be executed and let do his job
+3) The archive contain a complex hierarchy that need do to be extracted and add specific folder to PATH env var
+
+## Solution Design
+
+A new installation flow need to be created to execute different installation strategies regarding these three cases above.
+
+First thing first, the archive need to be extracted in a specific path (TEMP folder can be a good candidate).
+
+Regarding case 1, installation flow check if TEMP folder contain a single binary and repack it
+into an MSI package (including setting PATH env var). Then, MSI installation flow can take control.
+
+Regarding case 2, installation flow check if TEMP folder contain an installer (.exe, .msi, .msix) and pass control
+to MSI/Other installation flow.
+
+Regarding case 3, installation flow check if TEMP folder does not contain an installer nor a single binary.
+This case is the complex one, example with 'vagrant' hierarchy:
+- root
+  - bin 
+    - vagrant.exe
+    - other.exe
+  - embedded
+    - bin
+    - some other folders
+
+This hierarchy can be packed into an MSI installer, but how can we specify which binary to add 
+in PATH env var (in that case bin/vagrant.exe)?
+
+Take a look into the following manifest examples (v1.1.0)
+
+### Case 1|2 YAML manifest
+
+```
+PackageIdentifier: Editor.App
+PackageVersion: 1.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
+Scope: ...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+ManifestType: installer
+ManifestVersion: 1.0.0
+```
+
+This is compliant with the actual schema v1.1.0
+
+### Case 3 YAML manifest
+
+```
+PackageIdentifier: Editor.App
+PackageVersion: 1.0
+MinimumOSVersion: 10.0.0.0
+InstallerType: zip
+Scope: ...
+InstallModes: ...
+InstallerSwitches:
+  Silent: ...
+  SilentWithProgress: ...
+UpgradeBehavior: install
+Installers:
+- Architecture: x64
+  InstallerUrl: https://editor.com/app/1.0.zip
+  InstallerSha256: ...
+  BinaryPath: bin/binary.exe
+ManifestType: installer
+ManifestVersion: 1.0.0
+```
+
+This is not compliant with the actual schema v1.1.0. We need to specify which binary to add into
+PATH env var. Therefore, schema need to be updated to take knowledge of that information.
+
+### Repack ZIP into MSI package
+
+This section need more specifications. There are many tool to repack an MSI package, some expansive and
+some apparently free (Advanced Installer ?).
+
+## UI/UX Design
+
+One downside of this feature is the MSI repackaging will be executed on client side.
+Which mean for each installation the same process will be executed and remain the same for an 
+application x.y.z.
+
+But with a proper MSI repacking solution, this operation should be transparent to the end-user.
```

Comment:
We would likely have some output for the user with the packaging step so they at least get an indication of progress.